qphix_interface.cpp

```diff
@@ -949,16 +960,25 @@ int invert_eo_qphix(spinor * const Even_new,
          solver_params,
          compression);
     }
+#ifdef QPHIX_MIC_SOURCE
   } else if (precision < rsdTarget<float>::value) {
+#else
+  } else {
+#endif
     if (QPHIX_SOALEN > VECLEN_SP) {
       masterPrintf("SOALEN=%d is greater than the single prec VECLEN=%d\n", QPHIX_SOALEN,
                    VECLEN_SP);
       abort();
     }
     masterPrintf("# INITIALIZING QPHIX SOLVER\n");
     masterPrintf("# USING SINGLE PRECISION\n");
+    _initQphix(0, nullptr, qphix_input_By, qphix_input_Bz, qphix_input_NCores,
+           qphix_input_Sy, qphix_input_Sz,
+           qphix_input_PadXY, qphix_input_PadXYZ, qphix_input_MinCt,
+           compression, QPHIX_FLOAT_PREC);
+
     if (compress12) {
-      return invert_eo_qphix_helper<float, VECLEN_DP, QPHIX_SOALEN, true>
+      return invert_eo_qphix_helper<float, VECLEN_SP, QPHIX_SOALEN, true>
```

Comment:
I think we need this as well, but I'm not sure the reordering routines work properly for precisions other than double right now... They will have to be properly templatised, but there is this the issue that the precision used in tmlQCD may be different form the one used in QPhiX (or the interface for that matter).